vaccine_feed_ingest/runners/md/arcgis/normalize.py

```diff
@@ -0,0 +1,445 @@
+#!/usr/bin/env python
+
+import datetime
+import json
+import os
+import pathlib
+import re
+import sys
+from typing import List, Optional, Tuple
+
+from pydantic import ValidationError
+from vaccine_feed_ingest_schema import location as schema
+
+from vaccine_feed_ingest.utils.log import getLogger
+
+ALL_DAYS = (
+    schema.DayOfWeek.SUNDAY,
+    schema.DayOfWeek.MONDAY,
+    schema.DayOfWeek.TUESDAY,
+    schema.DayOfWeek.WEDNESDAY,
+    schema.DayOfWeek.THURSDAY,
+    schema.DayOfWeek.FRIDAY,
+    schema.DayOfWeek.SATURDAY,
+)
+DAY_OF_WEEK = r"(sun|mon|tues|wed|thurs|fri|sat)\.?"
+AM_PM = r"(a|p)\.?m\.?"
+TIME = r"\d{1,2}(:\d{1,2})?\s*" + AM_PM
+TIME_RANGE = TIME + r"\s*(-|to)\s*" + TIME
+
+# Configure logger
+logger = getLogger(__file__)
+
+
+def _get_access(site: dict) -> Optional[schema.Access]:
+    potentials = {
+        "Yes": True,
+        "No": False,
+    }
+    # TODO: check to see if the arcgis fields really map to Access this way
+    drive = potentials.get(site["attributes"]["drivethru"])
+    walk = potentials.get(site["attributes"]["pedaccess"])
+    if (drive, walk) != (None, None):
+        return schema.Access(drive=drive, walk=walk)
+    else:
+        return None
+
+
+# if anyone has a better way to handle these, I'm all ears.  just looked them
+# up on google maps, hardcoded the parseable version.
+SPECIAL_CASE_ADDRESSES = {
+    # missing zipcode
+    "12101 Winchester Road Sw Cumberland Allegany MD": "12101 Winchester Rd SW, Cumberland, MD 21502",
+    "1729 Dual Highway Hagerstown Washington MD": "1729 Dual Hwy, Hagerstown, MD 21740",
+    "18726 N. Pointe Drive Hagerstown Washington MD": "18726 N Pointe Dr, Hagerstown, MD 21742",
+    "12154 Brittingham Lane, Princess Anne MD": "12154 Brittingham Ln, Princess Anne, MD 21853",
+    "3935 Erdman Ave #37 Baltimore MD": "3935 Erdman Ave #37, Baltimore, MD 21213",
+    # missing state, and according to walgreens' site, it's the wrong zip
+    "9621 Belair Rd Perry Hall 21128": "9621 Belair Rd Baltimore, MD 21236",
+    # missing space between state and zip
+    "1300 E North Ave Baltimore MD21213": "1300 E North Ave Baltimore MD 21213",
+}
+ADDRESS_RE = re.compile(
+    r"^(?P<street>.*),? (?P<city>[\w \-']+),?\s+(?P<state>md|maryland),?\s+(?P<zip>\d{5}(-\d{4})?)$",
+    flags=re.I,
+)
+
+
+def _get_address(site: dict) -> schema.Address:
+    raw = site["attributes"]["fulladdr"]
+    processed = SPECIAL_CASE_ADDRESSES.get(raw, raw).strip()
+    if (
+        site["attributes"]["name"] == "Leidos Field at Ripken Ironbirds Stadium"
+        and processed == "At this address"
+    ):
+        # extra paranoia for the "at this address" address
+        processed = "873 Long Dr, Aberdeen, MD 21001"
+    match = ADDRESS_RE.search(processed)
+    if not match:
+        raise RuntimeError(f"Unable to parse address string '{raw}'")
```

Comment:
Could we update this `raise` (and the other similar ones) to fail more gracefully - perhaps `logger.warning`-ing but leaving this part of the schema empty?

Currently it looks like `poetry run vaccine-feed-ingest normalize md/arcgis` does not complete successfully because of this `raise`.

Reply:
Hell, that was there for me to be better able to fix errors as I found them, I meant to remove it before committing.  Cleaning it up.